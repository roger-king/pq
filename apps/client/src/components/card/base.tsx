import React, { useState } from 'react';
import { Card, Box, CardBody, Button, Text, Anchor, CardFooter, CardHeader, Heading } from 'grommet';
import { Bookmark, Chat } from 'grommet-icons';
import { VoteButtonGroupContainer } from '../../containers/voteBtnGroup.container';

interface BasicCardProps {
    postId: number;
    totalVotes: number;
    footer: string;
    header: string;
    body: string;
    didVote: { upvote: boolean; downvote: boolean };
    navigate?(): void;
}

export const BasicCard: React.FC<BasicCardProps> = (props: BasicCardProps) => {
    const { footer, header, body, postId, totalVotes, didVote, navigate } = props;
    const [showBorder, setShowBorder] = useState<boolean>(false);

    return (
        <Card
            background="white"
            round="15px"
            height={{ min: 'auto', max: '600px' }}
            overflow="none"
            margin={{ top: '20px' }}
            onClick={() => {
                if (navigate !== undefined) {
                    navigate();
                }
            }}
            onMouseOver={() => {
                setShowBorder(true);
            }}
            onMouseLeave={() => {
                setShowBorder(false);
            }}
            border={showBorder ? 'all' : false}
            style={{ cursor: navigate !== undefined ? 'pointer' : 'auto' }}
        >
            <CardHeader pad="medium">
                <Heading level="4" margin="none">
                    {header}
                </Heading>
            </CardHeader>
            <CardBody pad="medium" direction="row" align="start" gap="small">
                <Box width="90%">{body}</Box>
                <VoteButtonGroupContainer
                    postId={postId}
                    total={totalVotes}
                    userUpvote={didVote.upvote}
                    userDownVote={didVote.downvote}
                />
            </CardBody>
            <CardFooter
                pad={{ left: '20px', right: '20px', bottom: '20px' }}
                direction="row"
                justify="start"
                align="center"
            >
                <Button size="small" icon={<Chat size="small" />} label="comment" plain />
                {/* TODO: change fill of icon and text if saved */}
                <Button size="small" icon={<Bookmark size="small" />} label="save" plain />
                <Text size="0.8em">
                    posted by <Anchor>{footer}</Anchor>
                </Text>
            </CardFooter>
        </Card>
    );
};
