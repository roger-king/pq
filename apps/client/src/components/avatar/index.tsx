import { AvatarIcon } from './avatar';

export default AvatarIcon;
