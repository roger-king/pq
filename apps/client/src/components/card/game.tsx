import React, { PropsWithChildren } from 'react';
import { Box } from 'grommet';
import { Timer } from '../timer';
import { ConnectionStatus } from '../connectionStatus';

export interface GameCardProps {
  time: number;
  connected: boolean;
  extendedFooter?: JSX.Element;
}

export const GameCard: React.FC<PropsWithChildren<GameCardProps>> = ({
  time,
  connected,
  children,
  extendedFooter,
}: PropsWithChildren<GameCardProps>) => {
  return (
    <Box background="light-4" pad="medium" width="70%" height="50%" border>
      <Box direction="row" height="20%" align="center" justify="center" gap="small">
        <Timer time={time} />
      </Box>
      <Box direction="column" height="72%" align="center" justify="center">
        {children}
      </Box>
      <Box direction="row" height="8%" align="center" justify="between">
        <ConnectionStatus connected={connected} />
        {extendedFooter}
      </Box>
    </Box>
  );
};
