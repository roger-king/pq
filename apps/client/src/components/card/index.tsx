export * from './question';
export * from './game';
