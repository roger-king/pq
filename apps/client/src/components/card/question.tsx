import React, { PropsWithChildren } from 'react';
import { Box } from 'grommet';

export interface QuestionCardProps {
  elevation?: boolean;
}

export const QuestionCard: React.FC<PropsWithChildren<QuestionCardProps>> = ({
  children,
  elevation,
}: PropsWithChildren<QuestionCardProps>) => {
  return (
    <Box elevation={elevation ? 'medium' : 'none'} pad="medium" border="all" height={{ min: '278px', max: '400px' }}>
      {children}
    </Box>
  );
};
