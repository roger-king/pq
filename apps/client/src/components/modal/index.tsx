import React, { PropsWithChildren } from 'react';
import { Box, Heading, Layer } from 'grommet';

export interface BaseModalProps {
  title: string;
  onClose(): void;
}

const Modal: React.FC<PropsWithChildren<BaseModalProps>> = (props: PropsWithChildren<BaseModalProps>) => {
  const { children, title } = props;

  return (
    <Layer modal>
      <Box fill align="center" justify="center" pad="medium" background="brand">
        <Box direction="row" align="center" justify="between" fill>
          <Heading level="3">{title}</Heading>
        </Box>
        {children}
      </Box>
    </Layer>
  );
};

export default Modal;
