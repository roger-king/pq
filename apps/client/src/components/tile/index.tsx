import { Tile } from './tile.component';

export default Tile;
