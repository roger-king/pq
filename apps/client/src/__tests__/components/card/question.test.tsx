import React from 'react';
import renderer from 'react-test-renderer';

import { QuestionCard } from '../../../components/card';

describe('<QuestionCard />', () => {
  test('should display QuestionCard', async () => {
    const tree = renderer
      .create(
        <QuestionCard>
          <div>testing</div>
        </QuestionCard>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
