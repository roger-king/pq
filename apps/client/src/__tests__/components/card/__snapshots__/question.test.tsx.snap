// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionCard /> should display QuestionCard 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 jlHNLW"
>
  <div>
    testing
  </div>
</div>
`;
